server/storage/types.go

```diff
@@ -4,7 +4,29 @@ import "github.com/docker/notary/tuf/data"
 
 // MetaUpdate packages up the fields required to update a TUF record
 type MetaUpdate struct {
-	Role    data.RoleName
-	Version int
-	Data    []byte
+	Role     data.RoleName
+	Version  int
+	Data     []byte
+	Channels []*Channel
```

Comment:
>  Is the case for writing to both staging and a particular key ID? (I was a little fuzzy on how that would work for threshold signing).

Yeah, that was the plan. You upload metadata and it gets put into channels based on the keys it's signed with, then users can coordinate about which ones to promote by picking the versions they want to sign. Then you can get "all staged" by just querying the staging channel, or a particular staged version by querying for (staged, key_id).

> (as opposed to the Get* functions, for which if multiple channels are passed only the data from the first channel that gun and role are in will be returned), 

I think you're just seeing the deficiencies of the in-memory implementation. For testing purposes I didn't think it was necessary to try to replicate everything that sql gives you, but maybe it is worth the effort?

The SQL implementation gives you what you would expect from the interfaces: when you `Get` from a set of channels, the file must exist in all of them, and if you `Update` to a set of channels, the metadata will be placed in all of them.